sql/driver/string_scanner.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tamir Duberstein (tamird@gmail.com)
+
+package driver
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+)
+
+const (
+	dateFormat                    = "2006-01-02"
+	timestampWithOffsetZoneFormat = "2006-01-02 15:04:05.999999999-07:00"
+)
+
+var _ sql.Scanner = &StringScanner{}
+var _ fmt.Stringer = &StringScanner{}
+
+// StringScanner is a thin wrapper around sql.NullString that supports
+// scanning from time.Time.
+type StringScanner struct {
+	nullString sql.NullString
```

Comment:
> Yeah, if we don't expose Valid then it's confusing to call it NullString.

Sure, I'll rename it.

> This implementation is very narrowly useful, only in situations where we don't actually care whether the string was valid and we just want to print something.

What does it mean for the string to be valid? In `sql.NullString`, `Valid` means "not-NULL". This implementation is useful beyond printing _something_: it's also the point of disambiguation between `DDate` and `DTimestamp`.

> I'm surprised that the go driver implements implicit conversions between (some) types and string. I don't think creating a type with even more implicit conversions is the right solution here. I think it would be better for the CLI and logic_test to scan into a slice of interface pointers, and then have a formatting function to convert whatever the driver gives us to a string.

That seems fine, but I don't understand why it's better; in the end, the behaviour will be the same. Your version will be more explicit, but that explicitness will be non-trivial only in the cases that are explicit in this implementation too.